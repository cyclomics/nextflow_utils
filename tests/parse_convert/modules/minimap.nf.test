nextflow_process {

    name "Test Process Minimap2AlignAdaptiveParameterized"
    script "parse_convert/modules/minimap.nf"
    process "Minimap2AlignAdaptiveParameterized"
    
    
    test("Minimap test output") {

        when {
            process {
                """
                fq_file = file("$baseDir/example_datasets/example_consensus_reads/reads/ABC0001_0.fastq")
                ref = file("$baseDir/example_datasets/reference/pJPVCS_BBCS.mmi")

                input[0] = Channel.of(fq_file)
                input[1] = Channel.of(ref)
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()

            // with(process.out[0]) {
            //     assert path(get(0)[1]).md5 == "6dff2faaa2f7a1ae3b1d568c71f26ed8"
            // }
        }
    }
}
