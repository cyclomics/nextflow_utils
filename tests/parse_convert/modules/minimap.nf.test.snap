{
    "Minimap test output": {
        "content": [
            {
                "0": [
                    [
                        "SOME_ID_1",
                        "ABC0001_0",
                        "ABC0001_0.bam:md5,31760b2437fedec99dc265bf24977f49"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-02-05T15:24:52.921717"
    }
}